src/test/java/com/answerdigital/academy/answerking/service/CategoryServiceTest.java

```diff
@@ -261,44 +259,60 @@ void testRemoveItemThatIsNotInCategory() {
     @Test
     void testRemoveItemFromCategoryThatDoesNotExist() {
         // given
-        Item item = new Item("Coca cola", "This is a coke", BigDecimal.valueOf(2.00d), true);
-        item.setId(itemId);
+        Item item = new Item("Coca cola", "This is a coke", BigDecimal.valueOf(2.00d));
+        item.setId(ITEM_ID);
 
         // when
         doReturn(Optional.empty()).when(categoryRepository).findById(anyLong());
-        Exception exception = assertThrows(NotFoundException.class, () -> categoryService.removeItemFromCategory(categoryId, itemId));
+        Exception exception = assertThrows(NotFoundException.class, () -> categoryService.removeItemFromCategory(CATEGORY_ID, ITEM_ID));
 
         // then
         assertFalse(exception.getMessage().isEmpty());
         verify(categoryRepository).findById(anyLong());
     }
 
     @Test
-    void testDeleteCategory() {
+    void testRetireCategory() {
         // given
         Category category = new Category("Drinks", "Our selection of drinks");
-        category.setId(categoryId);
+        category.setId(CATEGORY_ID);
+
+        Category expectedCategory = new Category("Drinks", "Our selection of drinks");
+        expectedCategory.setId(CATEGORY_ID);
+        expectedCategory.setRetired(true);
 
         // when
-        doReturn(Optional.of(category)).when(categoryRepository).findById(anyLong());
+        doReturn(Optional.of(category))
+                .when(categoryRepository)
+                .findById(anyLong());
+        doReturn(expectedCategory)
+                .when(categoryRepository)
+                .save(any(Category.class));
+
+        category = categoryService.retireCategory(CATEGORY_ID);
 
         // then
-        assertDoesNotThrow(() -> categoryService.deleteCategoryById(categoryId));
+        assertEquals(expectedCategory, category);
         verify(categoryRepository).findById(anyLong());
+        verify(categoryRepository).save(any(Category.class));
     }
 
     @Test
-    void testDeleteCategoryThatDoesNotExist() {
+    void testRetireCategoryThatIsAlreadyRetiredThrowsRetirementException() {
```

Comment:
Could we also please get a couple of tests for `testRetireCategoryThatDoesNotExist` and `testRetireItemthatDoesNotExist` to check they will both produce a NotFoundException?